cli/remotebazel/remotebazel.go

```diff
@@ -1026,6 +973,104 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	return exitCode, nil
 }
 
+func attemptRun(ctx context.Context, bbClient bbspb.BuildBuddyServiceClient, execClient repb.ExecutionClient, req *rnpb.RunRequest) (*inpb.GetInvocationResponse, *repb.ExecuteResponse, error) {
+	childCtx, cancelChildCtx := context.WithCancel(ctx)
+	defer cancelChildCtx()
+
+	var inRsp *inpb.GetInvocationResponse
+	var execRsp *repb.ExecuteResponse
+
+	rsp, err := bbClient.Run(ctx, req)
+	if err != nil {
+		return nil, nil, status.UnknownErrorf("error running bazel: %s", err)
+	}
+
+	iid := rsp.GetInvocationId()
+	log.Debugf("Invocation ID: %s", iid)
+
+	// If the remote bazel process is canceled or killed, cancel the remote run
+	isInvocationRunning := true
+	go func() {
+		select {
+		case <-ctx.Done():
+		case <-childCtx.Done():
+		}
+
+		if !isInvocationRunning {
+			return
+		}
+
+		// Use a non-cancelled context to ensure the remote executions are
+		// canceled
+		_, err = bbClient.CancelExecutions(context.WithoutCancel(ctx), &inpb.CancelExecutionsRequest{
```

Comment:
even though the context is non-cancelled here, it doesn't look like anything waits for this RPC to complete? so when the CLI exits, the RPC could get interrupted and the executions won't get cancelled. Might want to do this in a `defer` instead of `go`